janusgraph-berkeleyje/src/main/java/org/janusgraph/diskstorage/berkeleyje/BerkeleyJEStoreManager.java

```diff
@@ -182,15 +195,17 @@ public BerkeleyJETx beginTransaction(final BaseTransactionConfig txCfg) throws B
             }
 
             return btx;
+        } catch (ThreadInterruptedException e) {
+            initialize();
```

Comment:
I am a bit confused. Why do we initialize again and then throw exception?

Reply:
`ThreadInterruptedException` exception means that the BerkeleyJE DB environment is no longer valid, `intialize()` will close and then reopen the `environment` so that subsequent transactions can work.

My reason for throwing here is because I think we should just leave a failed request as failed. And leave any retrying logic to the client.

Reply:
I have implemented the interruption retry logic. Let me know what you think.